### PR TITLE
refactor: OpenAPI import

### PR DESCRIPTION
**Pull request by bzp2010**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Part of refactoring OpenAPI support to provide OpenAPI implementation based on the previous `Loader` interface and modifying the import handler to support handling of `DataSets` imports.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/test/e2enew/data_loader/openapi3_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		Entry("petstore-expanded.yaml", func() {
+			path, err := filepath.Abs("../../testdata/import/petstore-expanded.yaml")
+			Expect(err).To(BeNil())
+
+			resp, err := client.R().
+				SetFile("file", path).
+				SetFormData(map[string]string{
+					"_file":     "default.json",
+					"type":      "openapi3",
+					"task_name": "test_petstore_expanded_yaml",
```
Comment (starsz) [inline api/test/e2enew/data_loader/openapi3_test.go:106]: Missing test for `MergeRoute `?

**Comment by nic-chen** [comment]
big PR...
will review later.

File: api/internal/handler/data_loader/loader/openapi3/import.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		data.Upstreams = append(data.Upstreams, upstream)
+	}
+
+	// each one will correspond to a route
+	for uri, v := range s.Paths {
+		realUri := regURIVar.ReplaceAllString(uri, "*")
+		routeID := o.TaskName + "_" + strings.NewReplacer("/", "-", "{", "", "}", "").Replace(strings.TrimPrefix(uri, "/"))
+
+		// decide whether to merge multi method routes based on configuration
+		if o.MergeRoute {
```
Comment (nic-chen) [inline api/internal/handler/data_loader/loader/openapi3/import.go:75]: We have a contributor who submitted a PR to merge routes with the same URI:
https://github.com/apache/apisix-dashboard/pull/2330

I think this feature should also be included after the refactoring:

File: api/internal/handler/data_loader/loader/openapi3/import.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+			// merge methods
+			for method := range v.Operations() {
+				route.Methods = append(route.Methods, strings.ToUpper(method))
+			}
+
+			// add upstream
+			if len(v.Servers) > 0 {
+				// create a new upstream when current path has separate Servers
+				upstreamID := routeID
+				upstream, path := generateUpstreamByServers(s.Servers, upstreamID)
```
Comment (nic-chen) [inline api/internal/handler/data_loader/loader/openapi3/import.go:89]: `s.Servers` or `v.Servers`?

**Pull request closed without merging**